pkg/storage/elasticsearch.go

```diff
@@ -38,6 +38,53 @@ type ElasticsearchDeployment struct {
 	Secrets    []corev1.Secret
 }
 
+func (ed *ElasticsearchDeployment) injectArguments(container *corev1.Container) {
+	container.Args = append(container.Args,
+		"--es.server-urls="+elasticsearchURL)
+	if util.FindItem("--es.tls=", container.Args) == "" {
+		container.Args = append(container.Args, "--es.tls=true")
+	}
+
+	container.Args = append(container.Args,
+		"--es.tls.ca="+caPath,
+		"--es.tls.cert="+certPath,
+		"--es.tls.key="+keyPath)
+
+	if util.FindItem("--es.timeout", container.Args) == "" {
+		container.Args = append(container.Args, "--es.timeout=15s")
+	}
+	if util.FindItem("--es.num-shards", container.Args) == "" {
+		// taken from https://github.com/openshift/cluster-logging-operator/blob/32b69e8bcf61a805e8f3c45c664a3c08d1ee62d5/vendor/github.com/openshift/elasticsearch-operator/pkg/k8shandler/configmaps.go#L38
+		// every ES node is a data node
+		container.Args = append(container.Args, fmt.Sprintf("--es.num-shards=%d", ed.Jaeger.Spec.Storage.Elasticsearch.NodeCount))
+	}
+	if util.FindItem("--es.num-replicas", container.Args) == "" {
+		container.Args = append(container.Args, fmt.Sprintf("--es.num-replicas=%d",
+			calculateReplicaShards(ed.Jaeger.Spec.Storage.Elasticsearch.RedundancyPolicy, int(ed.Jaeger.Spec.Storage.Elasticsearch.NodeCount))))
+	}
+	if strings.EqualFold(util.FindItem("--es-archive.enabled", container.Args), "--es-archive.enabled=true") {
+		container.Args = append(container.Args,
+			"--es-archive.server-urls="+elasticsearchURL,
+			"--es-archive.tls=true",
```

Comment:
Shouldn't this use the same override approach as for the `es.tls` parameter?

Reply:
For this test is not necessary, but yeah for consistency I can use the same approach.